### PR TITLE
Adjusted set_endtime method to default to UTC timestamps. Corrected m…

### PR DESCRIPTION
**Pull request by jpavlav**
…ispelling of jpavlav in doc/source/conf.py. We may need to discuss the change to set_endtime, but I think this is a safe bet for now unless we find otherwise.

### CODE REVIEW

**Comment by Kamori** [comment]
I don't mind defaulting by 0, but this function should definitely allow that to be passed through though so consumers can provide.

And to be clear timezone is offset by seconds according to chronyk while its nice to be able to type `'in 2 minutes'`

Does this feel pythonic? we can switch to delorean or maya instead if it makes sense

**Comment by Kamori** [comment]
Also, you mentioned the idea of checking the servers timezone first. Need to do more testing inside my docker container, but this might be easier than i thought.
```
[kamori@kamori-pc ~]$ curl -X OPTIONS 127.0.0.1:9093/api/v1/alerts -v
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 9093 (#0)
> OPTIONS /api/v1/alerts HTTP/1.1
> Host: 127.0.0.1:9093
> User-Agent: curl/7.59.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Access-Control-Allow-Headers: Accept, Authorization, Content-Type, Origin
< Access-Control-Allow-Methods: GET, DELETE, OPTIONS
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Date
< Date: Wed, 25 Apr 2018 02:07:16 GMT
< Content-Length: 0
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host 127.0.0.1 left intact
```


https://github.com/prometheus/alertmanager/blob/master/api/api.go#L66-L71

**Comment by jpavlav** [comment]
So, after doing a bit more research, it looks like AlertManager is supposed to default to UTC:

https://github.com/prometheus/alertmanager/issues/363

Having said that, I can't vouch for whether or not all versions of alert manager do this, so I think it makes sense to allow the user to decide what timezone they use to post their silences/alerts to alert manager. We should look into the possibility that the Alert.set_endtime() method should allow for the passing of a timezone.

The one thing that I think is rough about it is managing those offsets if they are in seconds. I'm a fan of passing strings that represent time into the method to get back the desired time. But I think for any kind of different timezone support, it might be worth it to look into other options. I'll delorean/maya and see about trying to get some tests implemented to figure out whether either of those are better options.

**Comment by jpavlav** [comment]
I motion, that we default to UTC, no current support for other timezones, with the possibility of it in the future. First, AlertManager expects rfc3339 format for dates/times:

```[
  {
    "labels": {
      "<labelname>": "<labelvalue>",
      ...
    },
    "annotations": {
      "<labelname>": "<labelvalue>",
    },
    "startsAt": "<rfc3339>",
    "endsAt": "<rfc3339>"
    "generatorURL": "<generator_url>"
  },
]
```

https://prometheus.io/docs/alerting/clients/

And the rfc docs suggest UTC as well:

```4.1. Coordinated Universal Time (UTC)

   Because the daylight saving rules for local time zones are so
   convoluted and can change based on local law at unpredictable times,
   true interoperability is best achieved by using Coordinated Universal
   Time (UTC).  This specification does not cater to local time zone
   rules.
```

https://www.ietf.org/rfc/rfc3339.txt

Additionally, I believe that `maya` is going to be our best bet for the `set_endtime` method. That update will be included in this PR.

File: alertmanager/alertmanager.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def add_annotation(self, key, value):
         self.annotations[key] = value
 
     def set_endtime(self, endtime):
-        self.endsAt = Chronyk(endtime).datetime().isoformat("T") + "Z"
+        self.endsAt = maya.when(endtime).rfc3339()
```
Comment (jpavlav) [inline alertmanager/alertmanager.py]: maya gives us the flexibility of passing in string arguments for end times with an auto-formatter for rfc3339

**Pull request merged**